drivers/modem/modem_cellular.c

```diff
@@ -2109,15 +2109,8 @@ MODEM_CHAT_SCRIPT_DEFINE(sqn_gm02s_periodic_chat_script,
 #define MODEM_CELLULAR_GET_USER_PIPES(inst) \
 	MODEM_CELLULAR_INST_NAME(user_pipes, inst)
 
-#define MODEM_CELLULAR_DEVICE_QUECTEL_BG95(inst)                                                   \
-	MODEM_PPP_DEFINE(MODEM_CELLULAR_INST_NAME(ppp, inst), NULL, 98, 1500, 64);                 \
-                                                                                                   \
-	static struct modem_cellular_data MODEM_CELLULAR_INST_NAME(data, inst) = {                 \
-		.chat_delimiter = "\r",                                                            \
-		.chat_filter = "\n",                                                               \
-		.ppp = &MODEM_CELLULAR_INST_NAME(ppp, inst),                                       \
-	};                                                                                         \
-                                                                                                   \
+/* Helper to define and initialize user pipes */
+#define MODEM_CELLULAR_DEFINE_AND_INIT_USER_PIPES(inst)                                            \
```

Comment:
The GNSS tunnelling DLCI channel us sometimes predefined. For the SARA-R4, it is channel 3, it could be channel 4 for another modem :) Could we add an `dlci_address_offset` and use it to set the `dlci_address` define user pipes macro
```
	MODEM_CELLULAR_DEFINE_USER_PIPES(                                                  
		inst,                                                         
		MODEM_CELLULAR_INIT_USER_PIPE(inst, gnss_pipe, dlci_address_offset),  
		MODEM_CELLULAR_INIT_USER_PIPE(inst, user_pipe_0, dlci_address_offset + 1),     
		MODEM_CELLULAR_INIT_USER_PIPE(inst, user_pipe_1, dlci_address_offset +2),   
	);   
```
or we can just name all pipes "user_pipe_<idx>" and maintain that user_pipe_0 == dlci channel 3 :) this will look better for modems which don't have an internal GNSS as well :)

Reply:
After a bit of feedback on discord, definitely just name them `user_pipe_<idx>` and have `user_pipe_0` and `user_pipe_1` :)

Reply:
Yes, we returned from the discussion on Discord. Also, modify `MODEM_CELLULAR_DEFINE_AND_INIT_USER_PIPES` to support variadic arguments in the recent pushed.